webassets_elm/__init__.py

```diff
@@ -1,70 +1,59 @@
-from contextlib import contextmanager
-import os
 import os.path
-from sys import platform
-from tempfile import NamedTemporaryFile, TemporaryFile
+import shutil
+from io import StringIO
+from tempfile import TemporaryDirectory
 
 from webassets.filter import ExternalTool
 
 __all__ = ('Elm')
 
 
-@contextmanager
-def excursion(directory):
-    """Context-manager that temporarily changes to a new working directory."""
-    old_dir = os.getcwd()
-    try:
-        os.chdir(directory)
-        yield
-    finally:
-        os.chdir(old_dir)
-
-
 class Elm(ExternalTool):
     """Elm filter for webassets.
 
     Requires the ``elm-make`` to be available from the system path.
 
     Supports the following external configuration:
 
-    ELM_MAKE_BIN
-        The path to the ``elm-make`` binary. If not set, assumes ``elm-make``
+    ELM_BIN (binary)
```

Comment:
I think the `_BIN` suffix already tells the user it's the binary, and the parenthesis as used before are mostly a type hint, so what about:

```suggestion
    ELM_BIN (string, path to the Elm binary)
```

Reply:
This was patterned on the existing webassets documentation, see the [`less` filter](https://webassets.readthedocs.io/en/latest/builtin_filters.html#webassets.filter.less.Less), and [`less_ruby`](https://webassets.readthedocs.io/en/latest/builtin_filters.html#less-ruby). Seeing this again now (both are basically the same plugin), I agree with you more than that precedent.

I'll just remove that postfix and leave it to the next line to explain the setting better, like it's done for the [`compass` plugin](https://webassets.readthedocs.io/en/latest/builtin_filters.html#webassets.filter.compass.Compass).